### PR TITLE
Upload logo to role

### PR DESCRIPTION
**Pull request by adudekyeti**
Upload logo to role

### CODE REVIEW

File: layouts/basic/modules/Settings/Roles/Index.tpl [hunk 1]
```diff
@@ -22,6 +22,9 @@
 					<div class="toolbar-handle">
 						<a href="javascript:;" class="btn btn-light draggable droppable">{\App\Language::translate($ROOT_ROLE->getName(), $QUALIFIED_MODULE)}</a>
 						<div class="toolbar" title="{\App\Language::translate('LBL_ADD_RECORD', $QUALIFIED_MODULE)}">
+							<button class="btn btn-light showModal ml-1 js-upload-logo" id="uploadLogo" type="button" data-js="click">
+								<span class="fas fa-image" title="{\App\Language::translate('LBL_SELECT',$MODULE_NAME)}"></span>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/Roles/Index.tpl]:
```suggestion
								<span class="fas fa-image" title="{\App\Language::translate('LBL_SELECT', $QUALIFIED_MODULE)}"></span>
```

File: layouts/basic/modules/Settings/Roles/Index.tpl [hunk 2]
```diff
@@ -22,6 +22,9 @@
 					<div class="toolbar-handle">
 						<a href="javascript:;" class="btn btn-light draggable droppable">{\App\Language::translate($ROOT_ROLE->getName(), $QUALIFIED_MODULE)}</a>
 						<div class="toolbar" title="{\App\Language::translate('LBL_ADD_RECORD', $QUALIFIED_MODULE)}">
+							<button class="btn btn-light showModal ml-1 js-upload-logo" id="uploadLogo" type="button" data-js="click">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/Roles/Index.tpl]:
```suggestion
							<button class="btn btn-light js-show-modal ml-1 js-upload-logo" id="uploadLogo" type="button" data-js="click">
```

File: modules/Settings/Roles/actions/UploadLogo.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Dudek <a.dudek@yetiforce.com>
+ */
+class Settings_Roles_UploadLogo_Action extends Settings_Vtiger_Index_Action
+{
+	public function __construct()
+	{
+		parent::__construct();
+		$this->exposeMethod('upload');
```
Comment (rskrzypczak) [inline modules/Settings/Roles/actions/UploadLogo.php]: One action. You don't need exposeMethod

File: modules/Settings/Roles/views/UploadLogo.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Dudek <a.dudek@yetiforce.com>
+ */
+
+/**
+ * Upload logo View Class.
+ */
+class Settings_Roles_UploadLogo_View extends Settings_Vtiger_BasicModal_View
```
Comment (rskrzypczak) [inline modules/Settings/Roles/views/UploadLogo.php]:
```suggestion
class Settings_Roles_UploadLogo_View extends \App\Controller\Modal
```

File: public_html/layouts/basic/modules/Settings/Roles/resources/Roles.js [hunk 5]
```diff
@@ -235,8 +235,69 @@ var Settings_Roles_Js = {
...
 		return aDeferred.promise();
 	},
-
+	/**
+	 * Register button to open modal logo upload form
+	 */
+	registerModalUploadButton: function () {
+		const thisInstance = this;
+		$('.js-upload-logo').on('click', function () {
+			AppConnector.request({
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Roles/resources/Roles.js]: ?

File: public_html/layouts/basic/modules/Settings/Roles/resources/Roles.js [hunk 6]
```diff
@@ -235,8 +235,69 @@ var Settings_Roles_Js = {
...
+				}
+			});
+		});
+	},
+	/**
+	 * Register button send form
+	 * @param container
+	 */
+	registerUploadButton: function (container) {
+		let form = container.find('.js-form-upload-logo');
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Roles/resources/Roles.js]: ?

File: public_html/layouts/basic/modules/Settings/Roles/resources/Roles.js [hunk 7]
```diff
@@ -235,8 +235,69 @@ var Settings_Roles_Js = {
...
+				}).done(function (data) {
+					progressIndicatorElement.progressIndicator({'mode': 'hide'});
+					if (true === data.result.success) {
+						Vtiger_Helper_Js.showPnotify({
+							title: app.vtranslate('JS_MESSAGE'),
+							text: app.vtranslate('JS_SAVE_NOTIFY_OK'),
+							type: 'success'
+						});
+					} else {
+						Vtiger_Helper_Js.showPnotify({
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Roles/resources/Roles.js]: one Vtiger_Helper_Js.showPnotify

File: modules/Settings/Roles/actions/UploadLogo.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Upload a logo.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Dudek <a.dudek@yetiforce.com>
+ */
+class Settings_Roles_UploadLogo_Action extends Settings_Vtiger_Index_Action
+{
+	public function process(\App\Request $request)
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Roles/actions/UploadLogo.php]: phpdoc

File: modules/Settings/Roles/actions/UploadLogo.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+	{
+		$targetFile = ROOT_DIRECTORY . DIRECTORY_SEPARATOR . 'public_html' . DIRECTORY_SEPARATOR . 'layouts' . DIRECTORY_SEPARATOR . 'resources' . DIRECTORY_SEPARATOR . 'Logo' . DIRECTORY_SEPARATOR . 'logo';
+		$response = new Vtiger_Response();
+		$result = ['success' => false, 'message' => \App\Language::translate('LBL_UPLOAD_ERROR', $request->getModule(false))];
+		if (!empty($_FILES['role_logo'])) {
+			$fileInstance = \App\Fields\File::loadFromPath($_FILES['role_logo']['tmp_name']);
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Roles/actions/UploadLogo.php]:
```suggestion
			$fileInstance = \App\Fields\File::loadFromRequest($_FILES['role_logo']);
```

**Pull request merged**